clients/python/girder_client/__init__.py

```diff
@@ -874,6 +874,57 @@ def uploadFileToItem(self, itemId, filepath, reference=None, mimeType=None, file
         with open(filepath, 'rb') as f:
             return self._uploadContents(obj, f, filesize, progressCallback=progressCallback)
 
+    def uploadFileToFolder(self, folderId, filepath, reference=None, mimeType=None, filename=None,
```

Comment:
@danlamanna Currently it doesn't allow that, but it would be easy to change.  We'd want to change both `uploadFileToItem` and the new `uploadFileToFolder`.  The internal function `_uploadContents` takes a file-like object, but doesn't have the logic for creating and updating items.  Also, if we aren't given a mime-type, we try to determine it from the path, so we would end up not being able to do this (though we could use the optional filename for that purpose).